{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import glob "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load TCR-Seq output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "TCR_files = glob.glob(\"../raw_data/TCR/*.tsv\")\n",
    "\n",
    "data = []\n",
    "\n",
    "for file in TCR_files:\n",
    "    dataframe = pd.read_csv(file,sep='\\t')\n",
    "    dataframe['sample'] = os.path.basename(file)\n",
    "    data.append(dataframe)\n",
    "\n",
    "TCR = pd.concat(data, sort=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# rename TCR sample names based on 1:1 matchup in python dictionary\n",
    "Sample_name_matchup_RNA_TCR_dict = {\n",
    "    \"ESO1-sorted-T-cells_S13_L007\" : \"RearrangementDetails_ESO1_sorted_infusion.tsv\",\n",
    "    \"INY1-sorted-T-cells_S14_L007\" : \"RearrangementDetails_INY1_sorted_infusion.tsv\",\n",
    "    \"INY2-sorted-T-cells_S15_L007\" : \"RearrangementDetails_INY2_sorted_infusion.tsv\",\n",
    "    \"TR2-PBMC_S12\" : \"RearrangementDetails_TR-PBMC.tsv\",\n",
    "    \"SAR-11-14-12RNA_S1\" : \"SAR_11-14_PBMC.tsv\",\n",
    "    \"MP-11-28-12RNA_S2\" : \"MP_11-28_PBMC.tsv\",\n",
    "    \"CMT-baseline1C_CAGATC\" : \"Pt204_Baseline_TCR_seq.tsv\",\n",
    "    \"HM-baseline1C_CGATGT\" : \"Pt310_baseline_TCRseq.tsv\",\n",
    "    \"PT0310_S9\" : \"Pt310_on-tx_TCR_seq.tsv\",\n",
    "    \"LEK-baseline_CGATGT\" : \"Pt294_baseline_TCR_seq.tsv\",\n",
    "    \"LEK-OT110712A_CCGTCC\" : \"Pt294_on-tx_TCR_seq.tsv\",\n",
    "    \"JSSBaseline-RNA_GTGAAA\" : \"Pt_308_baseline_TCR_seq.tsv\",\n",
    "    \"RAS-baseline_TGACCA\" : \"Pt_325_baseline_TCR_seq.tsv\",\n",
    "    \"PT0112-B_S3\" : \"Pt_112_baseline_TCR_seq.tsv\",\n",
    "    \"PT0285-B_S5\" : \"Pt_285_baseline_TCR_seq.tsv\"\n",
    "}\n",
    "\n",
    "for matchup in Sample_name_matchup_RNA_TCR_dict:\n",
    "    TCR.loc[(TCR['sample'] == Sample_name_matchup_RNA_TCR_dict[matchup]), 'sample' ] = matchup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/keruipeng/opt/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:376: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/Users/keruipeng/opt/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:494: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[item] = s\n"
     ]
    }
   ],
   "source": [
    "# rename and select the columns that needed for analysis\n",
    "TCR.rename(columns = {'reads':'reads_tool','rearrangement':'nucleotide','v_resolved':'V','d_resolved':'D','j_resolved':'J'},inplace=True)\n",
    "TCR = TCR[['reads_tool','V','J','sample']]\n",
    "TCR['tool'] = \"TCR-Seq\"\n",
    "\n",
    "# calculate frequency\n",
    "samples = ['CMT-baseline1C_CAGATC' , 'ESO1-sorted-T-cells_S13_L007',\n",
    "           'HM-baseline1C_CGATGT' , 'INY1-sorted-T-cells_S14_L007',\n",
    "           'INY2-sorted-T-cells_S15_L007' , 'JSSBaseline-RNA_GTGAAA',\n",
    "           'LEK-OT110712A_CCGTCC' , 'LEK-baseline_CGATGT' , 'MP-11-28-12RNA_S2',\n",
    "           'PT0112-B_S3' , 'PT0285-B_S5' , 'PT0310_S9' , 'RAS-baseline_TGACCA',\n",
    "           'SAR-11-14-12RNA_S1' , 'TR2-PBMC_S12']\n",
    "\n",
    "TCR_frequency = pd.DataFrame()\n",
    "\n",
    "for sample in samples:\n",
    "    sample_frequency = TCR.loc[(TCR['sample'] == sample)]\n",
    "    sample_frequency.loc[:,'frequency_tool'] = sample_frequency['reads_tool']/sum(sample_frequency['reads_tool'])\n",
    "    TCR_frequency = pd.concat([sample_frequency,TCR_frequency],sort=False)\n",
    "\n",
    "# select rows with reads greater than 1\n",
    "TCR_frequency = TCR_frequency[TCR_frequency['reads_tool'] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter out the unknown V genes and more than one V genes\n",
    "TCR_V = TCR_frequency[(TCR_frequency['V'] != 'unresolved')&(TCR_frequency['V'] != 'unknown')]\n",
    "\n",
    "# consider V gene only, not allele  \n",
    "TCR_V.loc[:,'V'] = TCR_V['V'].str.replace(\"(-).*\",\"\")\n",
    "\n",
    "# match the naming of gene \n",
    "TCR_V.loc[:,'V'] = TCR_V['V'].str.replace(\"TCRBV0\",\"TRBV\")\n",
    "TCR_V.loc[:,'V'] = TCR_V['V'].str.replace(\"TCRBV\",\"TRBV\")\n",
    "\n",
    "# calculate V gene usage \n",
    "TCR_V_frequency = TCR_V.groupby(['sample','V'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "TCR_V_frequency['tool'] = \"TCR-Seq\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter out the unknown J genes\n",
    "TCR_J = TCR_frequency[(TCR_frequency['J'] != 'unresolved')&(TCR_frequency['J'] != 'unknown')]\n",
    "\n",
    "# consider J gene only, not allele  \n",
    "TCR_J.loc[:,'J'] = TCR_J['J'].str.replace(\"(-).*\",\"\")\n",
    "\n",
    "# match the naming of gene \n",
    "TCR_J.loc[:,'J'] = TCR_J['J'].str.replace(\"TCRBJ0\",\"TRBJ\")\n",
    "TCR_J.loc[:,'J'] = TCR_J['J'].str.replace(\"TCRBJ\",\"TRBJ\")\n",
    "\n",
    "# calculate J gene usage \n",
    "TCR_J_frequency = TCR_J.groupby(['sample','J'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "TCR_J_frequency['tool'] = \"TCR-Seq\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load MIXCR output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "MIXCR_files = glob.glob(\"../raw_data/MIXCR/*.txt\")\n",
    "\n",
    "MIXCR = pd.DataFrame()\n",
    "for sample in MIXCR_files:\n",
    "    df_sample = pd.read_csv(sample, delimiter = \"\\t\")\n",
    "    for sample_name in Sample_name_matchup_RNA_TCR_dict.keys():\n",
    "        if sample_name in sample:\n",
    "            df_sample['sample'] = sample_name\n",
    "    MIXCR = MIXCR.append(df_sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename and select the columns that needed for analysis\n",
    "MIXCR['allVHitsWithScore']=MIXCR['allVHitsWithScore'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "MIXCR['allDHitsWithScore']=MIXCR['allDHitsWithScore'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "MIXCR['allJHitsWithScore']=MIXCR['allJHitsWithScore'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "MIXCR.rename(columns = {'cloneFraction':'frequency_tool','cloneCount':'reads_tool','aaSeqCDR3':'amino_acid','targetSequences':'nucleotide','allVHitsWithScore':'V','allDHitsWithScore':'D','allJHitsWithScore':'J'},inplace=True)\n",
    "MIXCR = MIXCR[['frequency_tool','reads_tool','V','J','sample']]\n",
    "MIXCR['tool'] = \"MIXCR\"\n",
    "\n",
    "# select TRB \n",
    "MIXCR = MIXCR[MIXCR['V'].str.contains(\"TRB\") & MIXCR['J'].str.contains(\"TRB\")]\n",
    "\n",
    "# select rows with reads greater than 1\n",
    "MIXCR = MIXCR[MIXCR['reads_tool'] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# consider V gene only, not allele  \n",
    "MIXCR['V'] = MIXCR['V'].str.replace(\"(-).*\",\"\") \n",
    "MIXCR['V'] = MIXCR['V'].str.replace(\"\\*.*\",\"\")\n",
    "\n",
    "# calculate V gene usage \n",
    "MIXCR_V_frequency = MIXCR.groupby(['sample','V'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "MIXCR_V_frequency['tool'] = \"MIXCR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# consider J gene only, not allele  \n",
    "MIXCR['J'] = MIXCR['J'].str.replace(\"(-).*\",\"\")\n",
    "MIXCR['J'] = MIXCR['J'].str.replace(\"\\*.*\",\"\")\n",
    "\n",
    "# calculate J gene usage \n",
    "MIXCR_J_frequency = MIXCR.groupby(['sample','J'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "MIXCR_J_frequency['tool'] = \"MIXCR\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load TRUST4 output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "TRUST4_files = glob.glob(\"../raw_data/TRUST4/*.tsv\")\n",
    "\n",
    "TRUST4 = pd.DataFrame()\n",
    "for sample in TRUST4_files:\n",
    "    df_sample = pd.read_csv(sample,sep='\\t')\n",
    "    for sample_name in Sample_name_matchup_RNA_TCR_dict.keys():\n",
    "        if sample_name in sample:\n",
    "            df_sample['sample'] = sample_name\n",
    "    TRUST4 = TRUST4.append(df_sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename and select the columns that needed for analysis\n",
    "TRUST4.rename(columns = {'frequency':'frequency_tool','#count':'reads_tool','CDR3aa':'amino_acid','CDR3nt':'nucleotide'},inplace=True)\n",
    "TRUST4 = TRUST4[['frequency_tool','reads_tool','V','J','sample']]\n",
    "TRUST4['tool'] = \"TRUST4\"\n",
    "\n",
    "# select TRB \n",
    "TRUST4 = TRUST4[TRUST4['V'].str.contains(\"TRB\") & TRUST4['J'].str.contains(\"TRB\")]\n",
    "\n",
    "# select rows with reads greater than 1\n",
    "TRUST4 = TRUST4[TRUST4['reads_tool'] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# consider V gene only, not allele  \n",
    "TRUST4['V'] = TRUST4['V'].str.replace(\"(-).*\",\"\") \n",
    "TRUST4['V'] = TRUST4['V'].str.replace(\"\\*.*\",\"\")\n",
    "\n",
    "# calculate V gene usage \n",
    "TRUST4_V_frequency = TRUST4.groupby(['sample','V'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "TRUST4_V_frequency['tool'] = \"TRUST4\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# consider J gene only, not allele  \n",
    "TRUST4['J'] = TRUST4['J'].str.replace(\"(-).*\",\"\")\n",
    "TRUST4['J'] = TRUST4['J'].str.replace(\"\\*.*\",\"\")\n",
    "\n",
    "# calculate J gene usage \n",
    "TRUST4_J_frequency = TRUST4.groupby(['sample','J'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "TRUST4_J_frequency['tool'] = \"TRUST4\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load IMREP output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "IMREP_files = glob.glob(\"../raw_data/IMREP/*.cdr3\")\n",
    "\n",
    "IMREP = pd.DataFrame()\n",
    "for sample in IMREP_files:\n",
    "    df_sample = pd.read_csv(sample)\n",
    "    for sample_name in Sample_name_matchup_RNA_TCR_dict.keys():\n",
    "        if sample_name in sample:\n",
    "            df_sample['sample'] = sample_name\n",
    "    IMREP = IMREP.append(df_sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename the columns \n",
    "IMREP.rename(columns = {'CDR3_AA_Seq':'amino_acid','Read_count':'reads_tool','V_chains':'V','D_chains':'D','J_chains':'J'},inplace=True)\n",
    "\n",
    "# select TRB \n",
    "IMREP = IMREP[IMREP['Chain_type'] == 'TRB']\n",
    "\n",
    "# calculate frequency\n",
    "IMREP['frequency_tool'] = IMREP['reads_tool']/sum(IMREP['reads_tool'])\n",
    "\n",
    "# select the columns that needed for analysis\n",
    "IMREP = IMREP[['frequency_tool','reads_tool','V','J','sample']]\n",
    "IMREP['tool'] = \"IMREP\"\n",
    "\n",
    "# select rows with reads greater than 1\n",
    "IMREP = IMREP[IMREP['reads_tool'] > 1]\n",
    "\n",
    "# drop the rows with more than one inferred V, J genes\n",
    "IMREP = IMREP[-IMREP['V'].str.contains(\";\")]\n",
    "IMREP = IMREP[-IMREP['J'].str.contains(\";\")]\n",
    "IMREP = IMREP[-IMREP['V'].str.contains(\"OR\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate V gene usage \n",
    "IMREP_V_frequency = IMREP.groupby(['sample','V'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "IMREP_V_frequency['tool'] = \"IMREP\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate J gene usage \n",
    "IMREP_J_frequency = IMREP.groupby(['sample','J'], as_index=False)['frequency_tool'].agg({'frequency_tool':'sum'})\n",
    "IMREP_J_frequency['tool'] = \"IMREP\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge dataframe "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>V</th>\n",
       "      <th>frequency_tool</th>\n",
       "      <th>tool</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>clonal_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBV1</td>\n",
       "      <td>0.000387</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBV10</td>\n",
       "      <td>0.069356</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBV11</td>\n",
       "      <td>0.034109</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBV12</td>\n",
       "      <td>0.002180</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBV13</td>\n",
       "      <td>0.004151</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>152</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBV30</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>153</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBV4</td>\n",
       "      <td>0.000560</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>154</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBV5</td>\n",
       "      <td>0.001732</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>155</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBV6</td>\n",
       "      <td>0.001110</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>156</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBV7</td>\n",
       "      <td>0.004185</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1058 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample       V  frequency_tool     tool sample_type  \\\n",
       "0    CMT-baseline1C_CAGATC   TRBV1        0.000387  TCR-Seq    melanoma   \n",
       "1    CMT-baseline1C_CAGATC  TRBV10        0.069356  TCR-Seq    melanoma   \n",
       "2    CMT-baseline1C_CAGATC  TRBV11        0.034109  TCR-Seq    melanoma   \n",
       "3    CMT-baseline1C_CAGATC  TRBV12        0.002180  TCR-Seq    melanoma   \n",
       "4    CMT-baseline1C_CAGATC  TRBV13        0.004151  TCR-Seq    melanoma   \n",
       "..                     ...     ...             ...      ...         ...   \n",
       "152           TR2-PBMC_S12  TRBV30        0.000473    IMREP        PBMC   \n",
       "153           TR2-PBMC_S12   TRBV4        0.000560    IMREP        PBMC   \n",
       "154           TR2-PBMC_S12   TRBV5        0.001732    IMREP        PBMC   \n",
       "155           TR2-PBMC_S12   TRBV6        0.001110    IMREP        PBMC   \n",
       "156           TR2-PBMC_S12   TRBV7        0.004185    IMREP        PBMC   \n",
       "\n",
       "    clonal_type  \n",
       "0    polyclonal  \n",
       "1    polyclonal  \n",
       "2    polyclonal  \n",
       "3    polyclonal  \n",
       "4    polyclonal  \n",
       "..          ...  \n",
       "152  polyclonal  \n",
       "153  polyclonal  \n",
       "154  polyclonal  \n",
       "155  polyclonal  \n",
       "156  polyclonal  \n",
       "\n",
       "[1058 rows x 6 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_V = pd.concat([TCR_V_frequency,MIXCR_V_frequency,TRUST4_V_frequency,IMREP_V_frequency],sort=False)\n",
    "\n",
    "# add sample type\n",
    "df_V.loc[df_V['sample']=='CMT-baseline1C_CAGATC','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='ESO1-sorted-T-cells_S13_L007','sample_type'] = 'PBMC'\n",
    "df_V.loc[df_V['sample']=='HM-baseline1C_CGATGT','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='INY1-sorted-T-cells_S14_L007','sample_type'] = 'PBMC'\n",
    "df_V.loc[df_V['sample']=='INY2-sorted-T-cells_S15_L007','sample_type'] = 'PBMC'\n",
    "df_V.loc[df_V['sample']=='JSSBaseline-RNA_GTGAAA','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='LEK-OT110712A_CCGTCC','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='LEK-baseline_CGATGT','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='MP-11-28-12RNA_S2','sample_type'] = 'PBMC'\n",
    "df_V.loc[df_V['sample']=='PT0112-B_S3','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='PT0285-B_S5','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='PT0310_S9','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='RAS-baseline_TGACCA','sample_type'] = 'melanoma'\n",
    "df_V.loc[df_V['sample']=='SAR-11-14-12RNA_S1','sample_type'] = 'PBMC'\n",
    "df_V.loc[df_V['sample']=='TR2-PBMC_S12','sample_type'] = 'PBMC'\n",
    "\n",
    "# add clonal type \n",
    "df_V.loc[df_V['sample']=='CMT-baseline1C_CAGATC','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='ESO1-sorted-T-cells_S13_L007','clonal_type'] = 'monoclonal'\n",
    "df_V.loc[df_V['sample']=='HM-baseline1C_CGATGT','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='INY1-sorted-T-cells_S14_L007','clonal_type'] = 'monoclonal'\n",
    "df_V.loc[df_V['sample']=='INY2-sorted-T-cells_S15_L007','clonal_type'] = 'monoclonal'\n",
    "df_V.loc[df_V['sample']=='JSSBaseline-RNA_GTGAAA','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='LEK-OT110712A_CCGTCC','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='LEK-baseline_CGATGT','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='MP-11-28-12RNA_S2','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='PT0112-B_S3','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='PT0285-B_S5','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='PT0310_S9','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='RAS-baseline_TGACCA','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='SAR-11-14-12RNA_S1','clonal_type'] = 'polyclonal'\n",
    "df_V.loc[df_V['sample']=='TR2-PBMC_S12','clonal_type'] = 'polyclonal'\n",
    "\n",
    "df_V.to_csv('../summary_data/150bp/all_tools_TRB_V_usage.csv', index=False)\n",
    "\n",
    "df_V"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>J</th>\n",
       "      <th>frequency_tool</th>\n",
       "      <th>tool</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>clonal_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBJ1</td>\n",
       "      <td>0.439541</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>CMT-baseline1C_CAGATC</td>\n",
       "      <td>TRBJ2</td>\n",
       "      <td>0.560148</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ESO1-sorted-T-cells_S13_L007</td>\n",
       "      <td>TRBJ1</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>monoclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ESO1-sorted-T-cells_S13_L007</td>\n",
       "      <td>TRBJ2</td>\n",
       "      <td>0.864367</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>monoclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>HM-baseline1C_CGATGT</td>\n",
       "      <td>TRBJ1</td>\n",
       "      <td>0.290288</td>\n",
       "      <td>TCR-Seq</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>RAS-baseline_TGACCA</td>\n",
       "      <td>TRBJ2</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>SAR-11-14-12RNA_S1</td>\n",
       "      <td>TRBJ1</td>\n",
       "      <td>0.000904</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>SAR-11-14-12RNA_S1</td>\n",
       "      <td>TRBJ2</td>\n",
       "      <td>0.000991</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBJ1</td>\n",
       "      <td>0.004334</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>TR2-PBMC_S12</td>\n",
       "      <td>TRBJ2</td>\n",
       "      <td>0.010490</td>\n",
       "      <td>IMREP</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>polyclonal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          sample      J  frequency_tool     tool sample_type  \\\n",
       "0          CMT-baseline1C_CAGATC  TRBJ1        0.439541  TCR-Seq    melanoma   \n",
       "1          CMT-baseline1C_CAGATC  TRBJ2        0.560148  TCR-Seq    melanoma   \n",
       "2   ESO1-sorted-T-cells_S13_L007  TRBJ1        0.041007  TCR-Seq        PBMC   \n",
       "3   ESO1-sorted-T-cells_S13_L007  TRBJ2        0.864367  TCR-Seq        PBMC   \n",
       "4           HM-baseline1C_CGATGT  TRBJ1        0.290288  TCR-Seq    melanoma   \n",
       "..                           ...    ...             ...      ...         ...   \n",
       "22           RAS-baseline_TGACCA  TRBJ2        0.000073    IMREP    melanoma   \n",
       "23            SAR-11-14-12RNA_S1  TRBJ1        0.000904    IMREP        PBMC   \n",
       "24            SAR-11-14-12RNA_S1  TRBJ2        0.000991    IMREP        PBMC   \n",
       "25                  TR2-PBMC_S12  TRBJ1        0.004334    IMREP        PBMC   \n",
       "26                  TR2-PBMC_S12  TRBJ2        0.010490    IMREP        PBMC   \n",
       "\n",
       "   clonal_type  \n",
       "0   polyclonal  \n",
       "1   polyclonal  \n",
       "2   monoclonal  \n",
       "3   monoclonal  \n",
       "4   polyclonal  \n",
       "..         ...  \n",
       "22  polyclonal  \n",
       "23  polyclonal  \n",
       "24  polyclonal  \n",
       "25  polyclonal  \n",
       "26  polyclonal  \n",
       "\n",
       "[114 rows x 6 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_J = pd.concat([TCR_J_frequency,MIXCR_J_frequency,TRUST4_J_frequency,IMREP_J_frequency],sort=False)\n",
    "\n",
    "# add sample type\n",
    "df_J.loc[df_J['sample']=='CMT-baseline1C_CAGATC','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='ESO1-sorted-T-cells_S13_L007','sample_type'] = 'PBMC'\n",
    "df_J.loc[df_J['sample']=='HM-baseline1C_CGATGT','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='INY1-sorted-T-cells_S14_L007','sample_type'] = 'PBMC'\n",
    "df_J.loc[df_J['sample']=='INY2-sorted-T-cells_S15_L007','sample_type'] = 'PBMC'\n",
    "df_J.loc[df_J['sample']=='JSSBaseline-RNA_GTGAAA','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='LEK-OT110712A_CCGTCC','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='LEK-baseline_CGATGT','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='MP-11-28-12RNA_S2','sample_type'] = 'PBMC'\n",
    "df_J.loc[df_J['sample']=='PT0112-B_S3','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='PT0285-B_S5','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='PT0310_S9','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='RAS-baseline_TGACCA','sample_type'] = 'melanoma'\n",
    "df_J.loc[df_J['sample']=='SAR-11-14-12RNA_S1','sample_type'] = 'PBMC'\n",
    "df_J.loc[df_J['sample']=='TR2-PBMC_S12','sample_type'] = 'PBMC'\n",
    "\n",
    "# add clonal type \n",
    "df_J.loc[df_J['sample']=='CMT-baseline1C_CAGATC','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='ESO1-sorted-T-cells_S13_L007','clonal_type'] = 'monoclonal'\n",
    "df_J.loc[df_J['sample']=='HM-baseline1C_CGATGT','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='INY1-sorted-T-cells_S14_L007','clonal_type'] = 'monoclonal'\n",
    "df_J.loc[df_J['sample']=='INY2-sorted-T-cells_S15_L007','clonal_type'] = 'monoclonal'\n",
    "df_J.loc[df_J['sample']=='JSSBaseline-RNA_GTGAAA','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='LEK-OT110712A_CCGTCC','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='LEK-baseline_CGATGT','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='MP-11-28-12RNA_S2','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='PT0112-B_S3','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='PT0285-B_S5','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='PT0310_S9','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='RAS-baseline_TGACCA','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='SAR-11-14-12RNA_S1','clonal_type'] = 'polyclonal'\n",
    "df_J.loc[df_J['sample']=='TR2-PBMC_S12','clonal_type'] = 'polyclonal'\n",
    "\n",
    "\n",
    "df_J.to_csv('../summary_data/150bp/all_tools_TRB_J_usage.csv', index=False)\n",
    "\n",
    "df_J"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
